api/custom_auth/views.py

```diff
@@ -30,7 +32,21 @@ class CustomAuthTokenLoginWithMFACode(AuthTokenLoginWithMFACode):
     throttle_scope = "mfa_code"
 
 
-class ThrottledUserViewSet(UserViewSet):
+@api_view(["DELETE"])
+@permission_classes([IsAuthenticated])
+def delete_token(request):
+    Token.objects.filter(user=request.user).delete()
+    user_logged_out.send(
+        sender=request.user.__class__, request=request, user=request.user
+    )
+    return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class UserDeleteQuerySerializer(UserDeleteSerializer):
+    delete_orphan_organisations = serializers.BooleanField(default=False)
+
+
+class FFAdminUserViewSet(UserViewSet):
```

Comment:
I think this could all be simplified to just rewrite the `perform_destroy` method. Something like the following: 

```python
@method_decorator(
    decorator=swagger_auto_schema(query_serializer=DeleteCurrentUserQuerySerializer()),
    name="destroy",
)
class ThrottledUserViewSet(UserViewSet):
    throttle_scope = "signup"

    def get_throttles(self):
        """
        Used for throttling create(signup) action
        """
        throttles = []
        if self.action == "create":
            throttles = [ScopedRateThrottle()]
        return throttles

    def perform_destroy(self, instance):
        if self.request.query_params:
            serializer = DeleteCurrentUserQuerySerializer(
                data=self.request.query_params
            )
            serializer.is_valid(raise_exception=True)

            if serializer.validated_data.get("delete_orphan_organisations"):
                self.request.user.delete_orpan_organisations()

        super().perform_destroy(instance)
```